src/store/modules/layers.store.js

```diff
@@ -65,6 +51,22 @@ const getters = {
         }
         return visibleLayers
     },
+    /**
+     * Get current KML layer selected for drawing.
+     *
+     * That is the KML layer that will be used when the drawing mode is opened.
+     *
+     * When no KML layer is in active layers then null is returned.
+     *
+     * @returns {KMLLayer | null}
+     */
+    activeKmlLayer: (state) => {
+        const kmlLayers = state.activeLayers.filter((layer) => layer.type === LayerTypes.KML)
+        if (kmlLayers.length > 0) {
+            return kmlLayers[kmlLayers.length - 1]
+        }
+        return null
+    },
```

Comment:
You should here probably use "visibleLayers.filter" instead of "activeLayers.filter", as for now you cannot unselect a (legacy) layer anymore to start a drawing.

Reply:
I cannot use visible layers filter here, because the kml layer in drawing mode is on purpose left in the active layers with the invisible flag. This allow to keep the kml in the url and if the user press reload, then it keep its drawing (also it would lose its adminId but can easily continue working on his drawing with a new copy). To keep the kml in the url parameter without making it invisible but removing from layer stack when drawing, we would need another parameter. I'll check if I can achieve this easily without too much hacking.
Note if you have a legacy drawing, you can still remove it from the layers in order to do a new drawing. The legacy blocking is only done for the MVP and will be removed once we are live, so this is not an issue.
One drawback with the actual implementation is that when in drawing mode and press F5 the drawing is then in invisible mode instead of visible.

Reply:
We also need to keep this like this to keep the same behavior as in old viewer. In older viewer, as long as you don't manually change the URL, you always have only one kml drawing in the layer list even if you deselect the kml layer (making is invisible) and you go in the drawing menu it will overwrite/copy the de-selected layer and make it visible again when leaving the drawing.
With my last commit I ensure to have only one drawing in the layer list.